docs/user-guide/tls.md

```diff
@@ -1,162 +1,110 @@
-By default this feature is disabled
+The flag `--enable-ssl-passthrough` enables the SSL passthrough feature.
+By default this feature is disabled.
+
+This is required to enable passthrough backends in Ingress configurations.
+
+TODO: Improve this documentation.
 
 ## HTTP Strict Transport Security
 
-HTTP Strict Transport Security (HSTS) is an opt-in security enhancement specified through the use of a special response header. Once a supported browser receives this header that browser will prevent any communications from being sent over HTTP to the specified domain and will instead send all communications over HTTPS.
+HTTP Strict Transport Security (HSTS) is an opt-in security enhancement specified
+through the use of a special response header. Once a supported browser receives
+this header that browser will prevent any communications from being sent over
+HTTP to the specified domain and will instead send all communications over HTTPS.
 
-By default the controller redirects (301) to HTTPS if there is a TLS Ingress rule.
+HSTS is enabled by default.
 
-To disable this behavior use `hsts: "false"` in the configuration ConfigMap.
+To disable this behavior use `hsts: "false"` in the configuration [ConfigMap][ConfigMap].
 
 ## Server-side HTTPS enforcement through redirect
 
-By default the controller redirects (301) to `HTTPS` if TLS is enabled for that ingress. If you want to disable that behavior globally, you can use `ssl-redirect: "false"` in the NGINX config map.
+By default the controller redirects HTTP clients to the HTTPS port
+443 using a 308 Permanent Redirect response if TLS is enabled for that Ingress.
+
+This can be disabled globally using `ssl-redirect: "false"` in the NGINX [config map][ConfigMap],
+or per-Ingress with the `nginx.ingress.kubernetes.io/ssl-redirect: "false"`
+annotation in the particular resource.
 
-To configure this feature for specific ingress resources, you can use the `nginx.ingress.kubernetes.io/ssl-redirect: "false"` annotation in the particular resource.
+!!! tip
 
-When using SSL offloading outside of cluster (e.g. AWS ELB) it may be useful to enforce a redirect to `HTTPS` even when there is not TLS cert available. This can be achieved by using the `nginx.ingress.kubernetes.io/force-ssl-redirect: "true"` annotation in the particular resource.
+    When using SSL offloading outside of cluster (e.g. AWS ELB) it may be useful to enforce a
+    redirect to HTTPS even when there is no TLS certificate available.
+    This can be achieved by using the `nginx.ingress.kubernetes.io/force-ssl-redirect: "true"`
+    annotation in the particular resource.
 
 ## Automated Certificate Management with Kube-Lego
 
-[Kube-Lego] automatically requests missing or expired certificates from [Let's Encrypt] by monitoring ingress resources and their referenced secrets. To enable this for an ingress resource you have to add an annotation:
+!!! tip
+    Kube-Lego has reached end-of-life and is being
```

Comment:
We should refresh that part to only mention cert-manager.

Reply:
Sure, but I feel that's out of scope for this PR.